daemon/daemon_main.go

```diff
@@ -1184,9 +1184,12 @@ func initEnv(cmd *cobra.Command) {
 		}
 	}
 
-	if option.Config.EnableNodePort {
+	if option.Config.EnableNodePort &&
+		!(option.Config.EnableHostReachableServices &&
+			option.Config.EnableHostServicesTCP && option.Config.EnableHostServicesUDP) {
 		// We enable host reachable services in order to allow
 		// access to node port services from the host.
+		log.Info("Enabling host reachable services for UDP and TCP required by BPF NodePort.")
```

Comment:
I think the statement is a bit confusing to a user? Question might come up whether anything needs to be done from user side. Maybe `Auto-enabling host reachable services for UDP and TCP as required by BPF NodePort.`

Reply:
Thought that `Enabling ...` will be read as `Auto-enabling`. But I will add the clarification.